### PR TITLE
Fix Stateful commands

### LINKED ISSUE #75: Drive command does not correctly select index
As the list is now split into Available Passengers and Taken Passengers in two separate panes, the drive command does not correctly accept index and select passenger correctly

### PR DESCRIPTION
**Pull request by JoelHo**
Fixes #75. Stateful commands (Delete/Drive/Find) were using the unfiltered list.

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -150,7 +150,7 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
 - prefers typing to mouse interactions
 - is reasonably comfortable using CLI apps
 
-**Value proposition**: Eliminate the need for human interaction such as requiring HR personnel to manage to maintain social distancing
+**Value proposition**: Eliminate the need for human interaction such as requiring HR passengernel to manage to maintain social distancing
```
Comment (lyueyang) [inline docs/DeveloperGuide.md]: wrong spelling for passenger

**Review by lyueyang — APPROVED** [review]
~~LGTM~~ approval gave for wrong PR

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (JoelHo) [reply docs/DeveloperGuide.md]: Fixed by commit 0c1b0db on PR #78

Comment (lyueyang) [reply docs/DeveloperGuide.md]: closing, resolved on improved code quality

**Review by ZechariahTan — APPROVED** [review]

**Review by ZechariahTan — APPROVED** [review]
LGTM

**Pull request merged**